bindings/Makefile

```diff
@@ -0,0 +1,45 @@
+PROJECT_ROOT := ..
+SWAGGER_SPEC := ${PROJECT_ROOT}/proto/build/swagger/determined/api/v1/api.swagger.json
+CODEGEN_BIN := swagger-codegen-cli-2.4.14.jar
+
+.PHONY: all
+all: get-deps build-proto
+	$(MAKE) get-swagger-spec
+	$(MAKE) build
+
+.PHONY: get-deps
+get-deps: deps/${CODEGEN_BIN}
+
+deps/${CODEGEN_BIN}:
+	mkdir -p deps
+	curl https://repo1.maven.org/maven2/io/swagger/swagger-codegen-cli/2.4.14/${CODEGEN_BIN} \
+		-o deps/${CODEGEN_BIN}
+
+build/%: ${SWAGGER_SPEC}
+	[ -f ./${SWAGGER_SPEC} ] || exit 1
+	mkdir -p build/$*
+	java -jar deps/$(CODEGEN_BIN) generate -i ${SWAGGER_SPEC} \
+		-l $* -o build/$*
+	touch build/$*
+
+.PHONY: build-default
+build-default: build/typescript-fetch build/python
+
+.PHONY: build
+build: ${SWAGGER_SPEC}
+	$(MAKE)	build-default
```

Comment:
why is this recursively calling make rather than just depending on `build-default`?

Reply:
initially, these two (build-default and ${SWAGGER_SPEC}) were not parallelizable

Reply:
Then make a chain of targets?